lib/matplotlib/tests/test_triangulation.py

```diff
@@ -58,14 +58,11 @@ def test_delaunay_duplicate_points():
     duplicate = 7
     duplicate_of = 3
 
-    np.random.seed(23)
-    x = np.random.random(npoints)
-    y = np.random.random(npoints)
-    x[duplicate] = x[duplicate_of]
-    y[duplicate] = y[duplicate_of]
+    xy = np.random.random((npoints, 2))
```

Comment:
https://dev.azure.com/matplotlib/matplotlib/_build/results?buildId=8800&view=logs&j=a00bdc14-80a3-51a4-3934-86a314523526&t=1906b723-82f5-57cd-ebf1-a6ef0c620a3f&l=138

It looks like sometimes it picks 7 to drop and sometimes it picks 3 to drop.  Should confirm with @ianthomas23 that is is OK behavior.

Reply:
That's exactly the kind of things that I was hoping to catch...

Reply:
*Short answer*:
The `seed` call should stay.

*Long answer*:
If we are using truly random (i.e. non-seeded) points, the qhull-generated delaunay triangulation algorithm will use just one of the two duplicated points, and we don't know in advance which one it will be. So there are two options:

1. Keep the `seed` call.

1. Remove the `seed` call and change the `assert` at the end of the test because the indices of the points used in the triangulation that are returned by qhull, `np.unique(triang.triangles)`, will either be `np.delete(np.arange(npoints), duplicate)` (the existing code) or `np.delete(np.arange(npoints), duplicate_of)`.

The first option is better for 3 reasons:
* It is simpler (less code).
* It is what the code currently does and existing working code shouldn't be changed without good reason.
* It is far better defensive programming to keep all tests deterministic. There is an exceedingly small but finite chance that a truly random (non-seeded) point will be close enough to the real duplicate points that the triangulation algorithm identifies this as the same point as the duplicate points and the test will fail with a different point index. The failure will happen at some point in the future on an entirely unrelated PR and will be totally unreproducible.

Hence keep the `seed` call. Then there is no reason to change any other code in the same function in a PR about `seed` calls.

There is a wider issue here that all tests should be deterministic and hence no `seed` calls should be removed without the explicit permission of each person who wrote the initial code in each case. Otherwise we are risking a number of extremely unlikely possible failures that are avoided with the existing code. The use of random numbers here is just shorthand for a small number of hard-coded points, we don't actually want to run the test with truly random numbers. If there is fundamental opposition to the use of `seed` in tests then the random values can be replaced with hard-coded ones. That is easily done for this test, but what about in `test_agg` that needs 70000 values?!

Reply:
I'm fine reverting the change here (which I will do) -- especially you point out to a real possible failure mode -- but actually, for most other cases, I am strongly convinced removing the seed is correct.  For eample, test_agg tests that you can write a 70,000-point line without raising an exception.  If there is *any* 70,000 point line which causes an exception, this is clearly a bug on our side.  It may be tricky to investigate, so we may patch back to forcing the seed while investigating, but we can't just say "oh, that's just an unlikely case, let's ignore it".  Likewise, tests involving check_figures_equal are testing logical equivalences -- two series of calls should result in the same backend-level draw operations, regardless of the input.